src/cpp/transport/TCPTransportInterface.cpp

```diff
@@ -449,6 +449,18 @@ bool TCPTransportInterface::transform_remote_locator(
     if (IsLocatorSupported(remote_locator) &&
         (!is_local_locator(remote_locator) || is_locator_allowed(remote_locator)))
     {
+        auto localInterface = current_interfaces_.begin();
+        for (; localInterface != current_interfaces_.end(); ++localInterface)
```

Comment:
This loop is already done inside is_local_locator, so the return value may be reused.

Looking at the code, I've discovered that is_local_locator should perform checks on the WAN address. Otherwise, we could interpret a remote LAN address as a local one.